components/_common_/Button/__tests__/Button.test.js

```diff
@@ -13,13 +13,15 @@ describe('Button', () => {
 
   test('should render properly with some props assigned', () => {
     createSnapshotTest(
-      <Button
-        theme="secondary"
-        data-custom-attr="custom stuff here"
-        disabled
-        fullWidth
-        type="submit"
-      >
+      <Button theme="secondary" disabled fullWidth type="submit">
+        Test
+      </Button>,
+    );
+  });
+
+  test('should spread data and aria props', () => {
```

Comment:
does the snapshot not verify the attribute was spread?

Reply:
fyi I'm new to snapshots. never had a need for them so let me know if i'm missing something.

Reply:
so a snapshot is good for regression tests and to ensure that the DOM reflects what you're trying to express.

To clarify what I mean, if you add a new prop or remove one (that's NOT aria or data), the test would fail. Aria and data could still be being passed.

Instead, you could test for the existence of the props with a shallow mount: https://airbnb.io/enzyme/docs/api/ShallowWrapper/prop.html

Reply:
Thanks, I understand. Does leaving the snapshot in place alongside the prop checking work?